proj4/src/main/scala/geotrellis/proj4/CRS.scala

```diff
@@ -1,13 +1,129 @@
+   * <li><b><tt>code</tt></b> is the id of a coordinate system in the authority namespace.
+   * For example, in the <tt>EPSG</tt> namespace a code is an integer value
+   * which identifies a CRS definition in the EPSG database.
+   * (Codes are read and handled as strings).
+   * </ul>
+   * An example of a valid CRS name is <tt>EPSG:3005</tt>.
+   * <p>
+   * @param name the name of a coordinate system, with optional authority prefix
+   * @return the [[CoordinateReferenceSystem]] corresponding to the given name
+   */
+  def fromName(name: String): CRS =
+    new CRS { val crs = crsFactory.createFromName(name)
+      override val epsgCode: Option[Int] = getEPSGCode(toProj4String+" <>")
+    }
+
+  /**
+   * Creates a [[CoordinateReferenceSystem]]
+   * from a PROJ.4 projection parameter string.
+   * <p>
+   * An example of a valid PROJ.4 projection parameter string is:
+   * <pre>
+   * +proj=aea +lat_1=50 +lat_2=58.5 +lat_0=45 +lon_0=-126 +x_0=1000000 +y_0=0 +ellps=GRS80 +units=m
+   * </pre>
+   * @param proj4Params a PROJ.4 projection parameter string
+   * @return the specified [[CoordinateReferenceSystem]]
+   */
+  def fromString(proj4Params: String): CRS =
+    new CRS { val crs = crsFactory.createFromParameters(null, proj4Params)
+      override val epsgCode: Option[Int] = getEPSGCode(toProj4String+" <>")
+    }
+
+  /**
+   * Creates a [[CoordinateReferenceSystem]]
+   * from a PROJ.4 projection parameter string.
+   * <p>
+   * An example of a valid PROJ.4 projection parameter string is:
+   * <pre>
+   * +proj=aea +lat_1=50 +lat_2=58.5 +lat_0=45 +lon_0=-126 +x_0=1000000 +y_0=0 +ellps=GRS80 +units=m
+   * </pre>
+   * @param name a name for this coordinate system.
+   * @param proj4Params a PROJ.4 projection parameter string
+   * @return the specified [[CoordinateReferenceSystem]]
+   */
+  def fromString(name: String, proj4Params: String): CRS =
+    new CRS{ val crs = crsFactory.createFromParameters(name, proj4Params)
+      override val epsgCode: Option[Int] = getEPSGCode(toProj4String+" <>")
+    }
+
+  /**
+   * Creates a [[CoordinateReferenceSystem]] (CRS) from a well-known-text String.
+   * @param wktString
+   * @return
+   */
+  def fromWKT(wktString :String): CRS ={
+    val epsgCode:String = WKT.getEPSGCode(wktString)
+
+    fromName(epsgCode)
+  }
+
+  /**
+   * Returns the numeric EPSG code of a proj4string
+   * @param proj4String
+   * @return
+   */
+  def getEPSGCode(proj4String: String):Option[Int]={
+     proj4ToEPSGMap(proj4String) match {
+       case Some(code) => Some(code.toInt)
+       case None =>  None
+     }
+  }
+
+  private def readEPSGCodeFromFile(proj4String: String): Option[String] ={
+    def code(line:String):Option[String]={
+      val array = line.split(" ")
+      val length = array(0).length
+      Some(array(0).substring(1,length-1))
+    }
+
+    Source.fromFile(filePrefix+"epsg").getLines().find(line=> !line.startsWith("#") && "+proj"+((line).split("proj"))(1)==proj4String) match {
```

Comment:
Format:

For math operators, we want spaces on either side. e.g.

`filePrefix + "epsg"`

BUT we shouldn't be using String `+` when it can be avoided (which is always).

Use string interpolation instead.

```scala
Source.fromFile(s"${filePrefix}epsg")
  .getLines
  .find { line => 
    val proj4Body = line.split("proj")(1)
    !line.startsWith("#") && s"+proj$proj4Body" == proj4String
   }
  .map { line => code(line) }
```
space around ` => `